### ISSUE TITLE
NoMethodError when removing a step from a campaign

### ISSUE THREAD

**Issue by rathboma** [honeybadger-io/heya#225]
I removed a step from a campaign, but now I get the following error in the background cron job:

NoMethodError: undefined method `action' for nil:NilClass

It's happening at this line: https://github.com/honeybadger-io/heya/blob/f472417b2126ad8986f93f9726a526b6b2e31ad1//lib/heya/campaigns/step_action_job.rb#L11

```ruby
      def perform(_campaign, user, step)
        step.action.new(user: user, step: step).deliver_now
      end
```
`step` is nil, so this fails.

**Comment by joshuap** [comment]
@rathboma thanks for reporting. Do you think there were some active jobs for that step when the change that removed it was deployed? We should probably add a guard for that. 🤔

**Comment by rathboma** [comment]
Yes, there could have been some active jobs that were not picked up properly after the restart.
